test/integration/execution/transaction.js

```diff
@@ -395,6 +395,30 @@ module.exports = function (knex) {
         });
     });
 
+    it('should save after transaction methods', function () {
+      let change1 = false;
+      let change2 = false;
+      const afterTransactionCollector = [];
+      return knex
+        .transaction(function (trx) {
+          trx.setOnAfterTransactionCollection(afterTransactionCollector);
+          trx.addFunctionOnAfterTransaction(async () => {
+            change1 = true;
+          });
+          trx.addFunctionOnAfterTransaction(async () => {
+            change2 = true;
+          });
+          trx.debugging = true;
+          return Promise.resolve(null);
+        })
+        .then(async function (result) {
+          await Promise.all(afterTransactionCollector.map((fn) => fn()));
```

Comment:
this way, users can decide what to do with these methods after transaction finishes